src/plugins/intel_gpu/src/kernel_selector/kernels/concatenation/concatenation_kernel_base.cpp

```diff
@@ -47,12 +47,38 @@ bool ConcatenationKernelBase::Validate(const Params& p, const optional_params&)
 }
 
 JitConstants ConcatenationKernelBase::GetJitConstants(const concatenation_params& params) const {
-    JitConstants jit = MakeBaseParamsJitConstants(params);
+    auto& inputs = params.original_input_layouts;
+    bool is_dynamic = std::any_of(inputs.begin(), inputs.end(), [](const DataTensor& t) { return t.is_dynamic(); }) ||
+                      std::any_of(params.outputs.begin(), params.outputs.end(), [](const DataTensor& t) { return t.is_dynamic(); });
```

Comment:
Can't we use params.is_shape_agnostic?

Reply:
Probably we can. Will refactor it in separate PR.